### PR TITLE
WGPU: Add new test and explanation for disabled test

### LINKED ISSUE #5612: Slang-RHI/WGPU: Invalid bind group layout entry used
Affected tests:

- `tests/bugs/buffer-swizzle-store.slang`

Example output:

    WGPU error: BindGroupLayoutEntry had none of buffer, sampler, texture, storageTexture, or externalTexture set
     - While validating entries[0]
     - While validating [BindGroupLayoutDescriptor]
     - While calling [Device].CreateBindGroupLayout([BindGroupLayoutDescriptor]).

### PR DESCRIPTION
**Pull request by aleino-nv**
- WGPU cannot support read-write access for storage textures of the format in the disabled test, so an explanation is added.
- A similar test is added, which uses structured buffers instead and should work also on WGSL.

See comments on https://github.com/shader-slang/slang/issues/5612 for more details.

This closes #5612.

### CODE REVIEW

File: tests/bugs/buffer-swizzle-store.slang [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj -output-using-type
 //TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type
 //TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl -output-using-type
-// Slang-RHI/WGPU: Invalid bind group layout entry used #5612
+// Not supported in WGSL: read-write storage texture with "rg16f" format
```
Comment (jkwak-work) [inline tests/bugs/buffer-swizzle-store.slang:4]: If the format is only problem for WGSL, can we change the format to one of formats WGSL supports?
Or maybe we can copy this test with a different format.

**Review by jkwak-work — APPROVED** [review]
Looks good to me

File: tests/bugs/buffer-swizzle-store.slang (hunk 1 above)
Comment (aleino-nv) [reply tests/bugs/buffer-swizzle-store.slang:4]: Yeah I can give it a shot.
There are [some Slang-RHI issues related to storage textures](https://github.com/shader-slang/slang/issues/5612#issuecomment-2514257576) I need to fix first though -- it currently doesn't create the bind group layout entry properly. I think I should set the `access` field to `read-write`, but I'm not quite sure how I should then get the required format information. I think I'll have to check in with Simon.

**Review by csyonghe — APPROVED** [review]

**Comment by csyonghe** [comment]
Needs another re-format.

**Review by jkwak-work — APPROVED** [review]

**Pull request merged**